docs/fundamentals/interface.rst

```diff
@@ -0,0 +1,105 @@
+
+Pipeline editing
+~~~~~~~~~~~~~~~~
+.. figure:: pipeline_example.png
+    :align: center
+    :width: 50%
+
+Pipelines are composed of a directed graph of nodes, where each node is an isolated python module. Nodes are added to a pipeline using the red plus button in the bottom right of the workspace. Any nodes that have previously been defined in the project can be added to the pipeline, or new operations can be created when needed. Arrows in the workspace indicate the passing of data between nodes. These arrows can be created by clicking on the desired output (bottom circles) of the first node before clicking on the desired input (top circles) of the second node. Clicking on a node also gives the options to delete (red X), edit (blue </>), or change attributes. Information on the editing of nodes can be found in `Custom Operations <custom_operations.rst>`_
+
+Pipelines are executed by clicking the yellow play button in the bottom right of the workspace. In the window that appears, you can name the execution, select a computation platform, and select a storage platform. The computation platform can either be SciServer's Compute service or a WebGME platform. The available storage platforms are SciServer's Files service and Amazon's S3 service. The provided storage option will be used for storing both the output objects defined in the pipeline, as well as all files used in execution of the pipeline. Login credentials will be required for SciServer computation service, either storage service, and each individual input node in the pipeline.
+
+.. figure:: execute_pipeline.png
+    :align: center
+    :width: 75%
+
+Executions
+----------
+.. figure:: executions_tab.png
+    :align: center
+    :width: 75%
+
+This view allows the review of previous pipeline executions. Clicking on any execution will display any plotted data generated by the pipeline, and selecting multiple executions will display all of the selected plots together. Clicking the provided links will open either the associated pipeline or a trace of the execution (shown below). The blue icon in the top right of every node allows viewing the text output of that node. The execution trace can be viewed during execution to check the status of a running job. During execution, the color of a node indicates its current status. The possible statuses are:
+
+- **Dark gray**: Awaiting initialization
+- **Light gray**: Awaiting execution
+- **Yellow**: Currently executing
+- **Green**: Successfully finished execution
+- **Red**: Execution failed
+
+.. figure:: execution_finished.png
+    :align: center
+    :width: 50%
+
+Resources
+---------
+.. figure:: resources_tab.png
+    :align: center
+    :width: 75%
+
+This view shows the available neural network resources available to your pipelines. From this view, resources can be created, deleted, and renamed. Resources are arranged by the deepforge system and cannot by manually reordered.
```

Comment:
Technically, neural network capabilities are not in the vanilla deepforge repo and are provided by the deepforge-keras extension. This would probably be worth mentioning. Maybe something like:

"This view shows the resources available for use in pipelines. Different types of resources are made available through DeepForge extensions and enable the introduction of new concepts into the project. One such example is `deepforge-keras` which enables users to make neural networks architectures with a custom visual editor. The created architectures can then be referenced and used by operations for tasks such as training."

Reply:
Changed. I also linked to the deepforge-keras repo for convenience.